### DISCUSSION TITLE
What's the validity of refresh_token

### DISCUSSION THREAD

**Discussion by janshermarri** [fief-dev/fief#72] [Q&A] [answered]
Can anyone please tell for how long the **refresh_token** is valid before it expires? Thanks.

**Comment by frankie567** [comment, marked as answer]
Hi @janshermarri, welcome to Fief 👋 

The `refresh_token` is valid for **30 days**. If you self-host your Fief instance, you can customize this parameter using the [`REFRESH_TOKEN_LIFETIME_SECONDS` environment variable](https://docs.fief.dev/self-hosting/environment-variables/#authorization-codes-and-tokens).

Bear in mind though that when you use the refresh token to get a fresh access token, you also get a **new refresh token**, that you should save for the future. The previous one is invalidated, you can use it only once. This is what is usually called **refresh token rotation**.
